neo/Network/P2P/Message.cs

```diff
@@ -1,79 +1,159 @@
-﻿using Neo.Cryptography;
+﻿using System;
+using System.IO;
+using Akka.IO;
+using Neo.Cryptography;
 using Neo.IO;
 using Neo.Network.P2P.Payloads;
-using System;
-using System.IO;
 
 namespace Neo.Network.P2P
 {
     public class Message : ISerializable
     {
-        public const int HeaderSize = sizeof(uint) + 12 + sizeof(int) + sizeof(uint);
         public const int PayloadMaxSize = 0x02000000;
+        public const int CompressionMinSize = 180;
+        public const int CompressionThreshold = 100;
 
-        public static readonly uint Magic = ProtocolSettings.Default.Magic;
-        public string Command;
-        public uint Checksum;
+        public MessageFlags Flags;
+        public MessageCommand Command;
         public byte[] Payload;
+        public short CheckSum;
```

Comment:
Move it to the front of `Payload`?

Reply:
For the consensus this `CheckSum` is not necessary, right? Because we already check the signature of the validator.

Reply:
Yes, it could be avoided in this type of messages.